test/lib/tmpdir.hh

```diff
@@ -60,5 +58,12 @@ public:
         remove();
     }
 
-    const fs::path& path() const noexcept { return _path; }
+    const fs::path& path() const & noexcept { return _path; }
+    // tmpdir().path() must not be allowed, use tmpdir::gen() for this.
+    const fs::path& path() const && = delete;
```

Comment:
I've never seen `= delete()` used like this. Makes sense though as C++ won't let you provide a `const&` overload specification unless there is also an `&&` one.

Reply:
It will allow you to write it so, but it's not helpful - the rvalue *this will decay to const& and call the const& overload.

Reply:
(so the = delete overload is necessary)

Reply:
I am wondering why
```
const fs::path& path() const & noexcept { return _path; }
```
allows `tmpdir().path()` if `&&` version is not deleted. spec as if says it must work without explicitly deleting `&&`.